From 4bc6cba6f19bd61d2c8124300deae99215a49196 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:49:49 -0400
Subject: [iommu] vt-d: Make identity_mapping() take struct device not struct pci_dev

Message-id: <20140905014949.31297.14190.stgit@gir.stowe>
Patchwork-id: 91333
O-Subject: [RHEL7.1 PATCH 40/65] iommu/vt-d: Make identity_mapping() take struct device not struct pci_dev
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 9b226624bbf7102cee67b6459bcb9c66dd081ca7

commit 9b226624bbf7102cee67b6459bcb9c66dd081ca7
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sun Mar 9 14:03:28 2014 -0700

    iommu/vt-d: Make identity_mapping() take struct device not struct pci_dev

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 5c89eb7..1030230 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2439,14 +2439,14 @@ static int __init si_domain_init(int hw)
  return 0;
 }
 
-static int identity_mapping(struct pci_dev *pdev)
+static int identity_mapping(struct device *dev)
 {
  struct device_domain_info *info;
 
  if (likely(!iommu_identity_mapping))
   return 0;
 
- info = pdev->dev.archdata.iommu;
+ info = dev->archdata.iommu;
  if (info && info != DUMMY_DEVICE_DOMAIN_INFO)
   return (info->domain == si_domain);
 
@@ -2903,7 +2903,7 @@ static int iommu_no_mapping(struct device *dev)
   return 0;
 
  pdev = to_pci_dev(dev);
- found = identity_mapping(pdev);
+ found = identity_mapping(dev);
  if (found) {
   if (iommu_should_identity_map(pdev, 0))
    return 1;
-- 
1.7.1